pkg/controller/factory/secret_creation.go

```diff
@@ -9,11 +9,14 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-// CreateOrUpdateSecretIfNotExists creates a secret in case it does not exist or updates it if there are changes
-func CreateOrUpdateSecretWithAPIReader(c client.Client, r client.Reader, secretName string, targetNS string, data map[string][]byte, secretType corev1.SecretType, log logr.Logger) error {
+// CreateOrUpdateSecretWithAPIReader creates a secret in case it does not exist or updates it if there are changes
+func CreateOrUpdateSecretWithAPIReader(c client.Client, r client.Reader, scheme *runtime.Scheme, owner metav1.Object, targetNS string, data map[string][]byte, secretType corev1.SecretType, log logr.Logger) error {
+	secretName := owner.GetName() + "-pull-secret"
```

Comment:
I'll presume this is the only one calling this function, but otherwise you'd want to send the name as a parameter. If that isn't the case, good for now.